### ISSUE TITLE
Private GitHub repos cannot be resolved when specifying user:token

### ISSUE THREAD

**Issue by sbglasius** [craigburke/client-dependencies-gradle#61]
It is not possible to specify 

```
bower {
    'private-assets'('1.0.0', url: 'https://user@token:github.com/private/private-assets.git')
}
```

Since https://github.com/craigburke/client-dependencies-gradle/blob/master/src/main/groovy/com/craigburke/gradle/client/registry/bower/GithubResolver.groovy#L25 cannot store `user` or `token` and use it for the API calls

**Comment by craigburke** [comment]
A solution to this would likely solve #44

**Comment by sbglasius** [comment]
There is apparently a little more to it, than that. My repo is not registred at https://bower.herokuapp.com (and should not be, since I want i private). 

Bower supports a structure like this:

```
"dependencies": {
    "my-assets": "https://github.com/myorg/my-assets.git#master"
}
```

But trying this with my repo being public:

```
bower {
    'my-assets'('1.0.1') {
        url '"https://github.com/myrepo/my-assets.git'
    }
}
```

Gives me this exception:

```
Caused by: java.io.FileNotFoundException: https://bower.herokuapp.com/packages/my-assets
        at com.craigburke.gradle.client.registry.bower.BowerRegistry.loadInfoFromRegistry(BowerRegistry.groovy:88)
        at com.craigburke.gradle.client.registry.core.AbstractRegistry$_setInfo_closure4.doCall(AbstractRegistry.groovy:146)
        at com.craigburke.gradle.client.registry.core.AbstractRegistry$_setInfo_closure4.doCall(AbstractRegistry.groovy)
```

(This is running on the latest 1.3.0-SNAPSHOT build locally)

**Comment by craigburke** [comment]
@sbglasius I need to update the docs, but I believe something like that is already supported. 

```
bower {
    'my-assets'('myrepo/my-assets#SHA')
}
```

You can currently use either the SHA hash for a particular commit or a tag (like `v1.0.0`).

I'm working on the basic authentication for github with user tokens now.

**Comment by craigburke** [comment]
Fixed with your PR #64. Released in version 1.3.0.
